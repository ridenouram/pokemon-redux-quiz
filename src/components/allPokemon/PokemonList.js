import React from 'react';
import PropTypes from 'prop-types';
import PokemonCard from './PokemonCard';

function PokemonList({ pokemonArr }) {
  const mappedPokemon = pokemonArr.map(pokemon => {
    return (
      <li key={pokemon._id}>
        <PokemonCard name={pokemon.pokemon} color={pokemon.color_1} img={pokemon.url_image} id={pokemon._id} />
      </li>
    );
  });

  return (
    <ul>
      {mappedPokemon}
    </ul>
  );
}

PokemonList.propTypes = {
  pokemonArr: PropTypes.array.isRequired
};

export default PokemonList;
