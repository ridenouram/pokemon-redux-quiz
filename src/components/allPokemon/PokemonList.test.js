import React from 'react';
import { shallow } from 'enzyme';
import PokemonList from './PokemonList';

describe('PokemonList component', () => {
  it('snapshot', () => {
    const wrapper = shallow(<PokemonList pokemonArr={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
