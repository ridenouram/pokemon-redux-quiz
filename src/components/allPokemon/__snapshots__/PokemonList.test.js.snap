// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonList component snapshot 1`] = `<ul />`;
